### PR TITLE
Use new new storage

### LINKED ISSUE #1544: Move extraction of data to new api

### PR DESCRIPTION
**Pull request by ManInFez**
**Issue**
Resolves #1544

### CODE REVIEW

**Comment by markusdregi** [comment]
I think well structured commits will make a review of this one much easier when that time comes 👏🏻

**Comment by ManInFez** [comment]
> I think well structured commits will make a review of this one much easier when that time comes 👏🏻

I think it is fine, just wait until I add a commit after the finalize commit 😅

**Comment by markusdregi** [comment]
Seems like there will be a `final finalisation` as well ;)

File: .github/workflows/run_ert2_test_data_setups.yml [hunk 1]
```diff
@@ -34,6 +34,7 @@ jobs:
     - name: Install ERT and dependencies
       run: |
         pip install .
+        pip install .[storage]
```
Comment (markusdregi) [inline .github/workflows/run_ert2_test_data_setups.yml:37]: Is `ert-storage` now needed to run the ert2 examples? 🤔

Comment (ManInFez) [reply .github/workflows/run_ert2_test_data_setups.yml:37]: it is a bit annoying, maybe we could discuss a workaround?

Comment (markusdregi) [reply .github/workflows/run_ert2_test_data_setups.yml:37]: Sure, I think at least that if `ert-storage` is needed to run ert2 there is no discussion anymore on whether it should be listed as one of the requirements... I guess the alternative is to go for a strategy where `ert-storage` only is imported at need / allowed to fail together with the feature flag only being accepted if `ert-storage` is available?

File: ert_shared/storage/extraction.py [hunk 2]
```diff
@@ -7,8 +7,10 @@
 from ert_shared.storage.server_monitor import ServerMonitor
 from res.enkf.export import MisfitCollector
```
Comment (oysteoh) [inline ert_shared/storage/extraction.py:8]: not in use anymore

File: ert_shared/storage/extraction.py [hunk 3]
```diff
@@ -7,8 +7,10 @@
...
 
-import ert_shared.storage.json_schema as js
 import requests
+import pandas as pd
+import io
```
Comment (oysteoh) [inline ert_shared/storage/extraction.py:12]: not in use anymore

File: ert_shared/storage/extraction.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-from typing import List, Optional, Mapping, Tuple
+from typing import List, Optional, Mapping, Tuple, Any, Dict
```
Comment (oysteoh) [inline ert_shared/storage/extraction.py:1]: Optional and Tuple not in use anymore

File: ert_shared/storage/extraction.py [hunk 5]
```diff
@@ -211,61 +229,104 @@ def _create_observation_transformation(ert):
...
-        data=create_ensemble(ert, update_id).json(),
-        auth=server.fetch_auth(),
+    if update_id is None:
+        exp_response = requests.post(
+            f"{server.fetch_url()}/experiments",
+            json=create_experiment(ert),
+        ).json()
+        experiment_id = exp_response["id"]
+        for obs in create_observations(ert):
+            obs_response = requests.post(
```
Comment (oysteoh) [inline ert_shared/storage/extraction.py:303]: obs_response is never used. Is there any reason to store it as a variable?

File: ert_shared/storage/extraction.py [hunk 6]
```diff
@@ -211,61 +229,104 @@ def _create_observation_transformation(ert):
...
+            parameter_names=[param["name"] for param in parameters],
+            update_id=update_id,
+        ),
     )
-    if not response.status_code == 200:
-        raise RuntimeError(response.text)
+    ensemble_id = ens_response.json()["id"]
+
+    for param in parameters:
+        param_response = requests.post(
```
Comment (oysteoh) [inline ert_shared/storage/extraction.py:324]: `param_response` is never used either

**Review by oysteoh — APPROVED** [review]
Only a couple of more comments - not sure if it needs to be adressed! Otherwise - looks very good, all tests green 🚀

**Pull request merged**